plugin/src/main/java/org/opensearch/ml/searchext/MLInferenceRequestParameters.java

```diff
@@ -0,0 +1,79 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.ml.searchext;
+
+import java.io.IOException;
+import java.util.Map;
+import java.util.Objects;
+
+import org.opensearch.core.common.io.stream.StreamInput;
+import org.opensearch.core.common.io.stream.StreamOutput;
+import org.opensearch.core.common.io.stream.Writeable;
+import org.opensearch.core.xcontent.ToXContentObject;
+import org.opensearch.core.xcontent.XContentBuilder;
+import org.opensearch.core.xcontent.XContentParser;
+
+import lombok.Getter;
+import lombok.NoArgsConstructor;
+import lombok.Setter;
+
+@NoArgsConstructor
+public class MLInferenceRequestParameters implements Writeable, ToXContentObject {
+    static final String ML_INFERENCE_FIELD = "ml_inference";
+
+    @Setter
+    @Getter
+    private Map<String, Object> params;
+
+    public MLInferenceRequestParameters(Map<String, Object> params) {
```

Comment:
if you add `final` to `param` field you can replace this constructor with Lombok's AllArgsContructor

Reply:
later during search processors, possibly, this params is going to get added with new value, so trying not to make it final

Reply:
The final keyword ensures that the reference to the map cannot be changed to point to another map, but it does not make the contents of the map immutable. You can still modify the contents, such as adding, updating, or removing items from the map.